### PR TITLE
Kactl convergence

### PR DESCRIPTION
**Pull request by RichardHladik**
The ability to add per-source-file header (containing such things as time/memory complexity, description of usage…) has been introduced. Relevant header fields are then typeset before the actual code listing.

### CODE REVIEW

File: export/build [hunk 1]
```diff
@@ -77,13 +77,35 @@ def highlight(text):
...
+        return text.replace("$O", "$\\O").replace("*", "\cdot ")
+
+    keywords = ["Time", "Memory", "Description", "Usage"]
+
+    l = []
+    for h in keywords:
+        if h in header:
+            if h == "Usage":
+                # TODO: quite a hack
+                l.append("{\\leftskip=\\usagedimen\\item{\\bf Usage:}\\simpleverb|%s\\egroup\\par}" % highlight(header[h]))
```
Comment (IAmWave) [inline export/build]: A raw string would improve readability here, allowing you to write `\` and not `\\`
```
r"\leftskip=\usage..."
```

File: export/build [hunk 2]
```diff
@@ -77,13 +77,35 @@ def highlight(text):
     return "".join(l)
 
 
+def make_header(header):
+    def texify(text):
+        return text.replace("$O", "$\\O").replace("*", "\cdot ")
```
Comment (IAmWave) [inline export/build]: ?

File: export/build (hunk 1 above)
Comment (IAmWave) [reply export/build]: Why is "usage" handled separately and is so different? (I see - it's a code snippet. Would be worth mentioning somewhere)

File: export/build [hunk 3]
```diff
@@ -117,4 +142,49 @@ def main():
 
     os.system("xetex %s" % OUT_NAME)
 
+
+def parse_header(text):
+    """Parses /** ... */ headers. Returns a tuple of Key: Content pairs."""
```
Comment (IAmWave) [inline export/build]: Maybe a doctest would illustrate usage nicely, but this is not high-priority

**Comment by RichardHladik** [comment]
I'm going to comment the denser parts in one of the next PRs, merging now since these are not major issues.

**Pull request merged**